src/main/java/Deadline.java

```diff
@@ -0,0 +1,43 @@
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a Task object with a set date to be completed by, denoted by the attribute LocalDate deadline.
+ */
+class Deadline extends Task {
+    private LocalDate deadline;
+
+    public Deadline(String task) throws DateTimeParseException {
+        // Split the input task string by the delimiter /by
+        // first element in arr will be the description of the task, second elem will be the deadline, both are String
+        // Pass the description as the argument to the constructor
+        super(task.split("/by")[0]);
+        String[] arr = task.split("/by");
+        try {
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-d");
+            this.deadline = LocalDate.parse(arr[1].trim(), formatter);
+        } catch (DateTimeParseException e) {
+            throw e;
+        }
+
+    }
```

Comment:
Model classes should not be handling exceptions - parsing should have already been handled by your Parser class, and assert that the parameters are valid here